pkg/login/social/social.go

```diff
@@ -21,72 +22,50 @@ var (
 	logger = log.New("social")
 )
 
-type BasicUserInfo struct {
-	Id      string
-	Name    string
-	Email   string
-	Login   string
-	Company string
-	Role    string
-	Groups  []string
+func init() {
+	registry.RegisterService(&Service{})
 }
 
-type SocialConnector interface {
-	Type() int
-	UserInfo(client *http.Client, token *oauth2.Token) (*BasicUserInfo, error)
-	IsEmailAllowed(email string) bool
-	IsSignupAllowed() bool
-
-	AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string
-	Exchange(ctx context.Context, code string, authOptions ...oauth2.AuthCodeOption) (*oauth2.Token, error)
-	Client(ctx context.Context, t *oauth2.Token) *http.Client
-	TokenSource(ctx context.Context, t *oauth2.Token) oauth2.TokenSource
+type Service struct {
+	Cfg          *setting.Cfg  `inject:""`
+	OAuthService *OAuthService `inject:""`
```

Comment:
I think we can skip the `OAuthService` field that only holds a map and let the `Service` have a non-exported `map[string]*OAuthInfo` together with some methods like get OAuth providers and get OAuth provider by name. WDYT?

Reason I don't like code like `hs.SocialService.OAuthService.OAuthInfos[name].<field>` being spread around the code case compared with 
```go
provider := hs.SocialService.GetOAuthProvider(name)
provider.<field1>
provider.<field2>
```

Reply:
I agree, this way is much nicer